cpp/include/Ice/LocalExceptions.h

```diff
@@ -323,897 +280,636 @@ namespace Ice
          * @param file The file where this exception is constructed. This C string is not copied.
          * @param line The line where this exception is constructed.
-         * @param kindOfObject The kind of object that could not be removed: "servant", "facet", "object", "default
-         * servant", "servant locator", "value factory", "plugin", "object adapter", "object adapter with router",
-         * "replica group".
-         * @param id The ID (or name) of the object that is registered already.
          */
-        AlreadyRegisteredException(const char* file, int line, std::string kindOfObject, std::string id) noexcept
-            : LocalException(file, line),
-              kindOfObject(std::move(kindOfObject)),
-              id(std::move(id))
-        {
-        }
+        InvocationTimeoutException(const char* file, int line) : TimeoutException(file, line, "invocation timed out") {}
 
-        /**
-         * Obtains a tuple containing all of the exception's data members.
-         * @return The data members in a tuple.
-         */
-        std::tuple<const std::string&, const std::string&> ice_tuple() const noexcept
-        {
-            return std::tie(kindOfObject, id);
-        }
-
-        const char* ice_id() const noexcept override;
-
-        void ice_print(std::ostream& stream) const override;
-
-        /**
-         * The kind of object that could not be removed: "servant", "facet", "object", "default servant",
-         * "servant locator", "value factory", "plugin", "object adapter", "object adapter with router", "replica
-         * group".
-         */
-        std::string kindOfObject;
-        /**
-         * The ID (or name) of the object that is registered already.
-         */
-        std::string id;
+        const char* ice_id() const noexcept final;
     };
 
+    //
+    // Syscall exceptions
+    //
+
     /**
-     * An attempt was made to find or deregister something that is not registered with the Ice run time or Ice locator.
-     * This exception is raised if an attempt is made to remove a servant, servant locator, facet, value factory,
-     * plug-in, object adapter, object, or user exception factory that is not currently registered. It's also raised if
-     * the Ice locator can't find an object or object adapter when resolving an indirect proxy or when an object adapter
-     * is activated. \headerfile Ice/Ice.h
+     * This exception is raised if a system error occurred in the server or client process. There are many possible
+     * causes for such a system exception. For details on the cause, {@link SyscallException#error} should be inspected.
+     * \headerfile Ice/Ice.h
      */
-    class ICE_API NotRegisteredException : public LocalException
+    class ICE_API SyscallException : public LocalException
     {
     public:
-        using LocalException::LocalException;
+        // The error code is a DWORD (unsigned long) on Windows and an int on other platforms.
```

Comment:
That's another benefit of dropping local Slice: the error code for Windows can have the correct type.